src/test/test_controller.c

```diff
@@ -1689,6 +1695,62 @@ test_download_status_bridge(void *arg)
   return;
 }
 
+tor_mmap_t * mock_networkstatus_map_cached_consensus(const char*);
+
+tor_mmap_t *
+mock_networkstatus_map_cached_consensus(const char *flavorname)
+{
+  tor_mmap_t *res = tor_malloc_zero(sizeof(tor_mmap_t));
+  if (!strcmp(flavorname, "ns")) {
+    res->data = "mock_ns_consensus";
+  }
+  if (!strcmp(flavorname, "microdesc")) {
+    res->data = "mock_microdesc_consensus";
+  }
+  res->size = strlen(res->data);
+  return res;
+}
+
+static void
+test_getinfo_helper_dir(void *arg)
+{
+  /* We just need one of these to pass, it doesn't matter what's in it */
+  control_connection_t dummy;
+  /* Get results out */
+  char *answer = NULL;
+  const char *errmsg = NULL;
```

Comment:
This unittest is failing in Appveyor for Windows. Any idea why? It might be that the extra free is causing trouble, or something else.

Reply:
It is something else :( I haven't been able to find what exactly. Without the extra free here the CI was failing on both Travis and Appveyor due to memory leaks.